releasenotes/notes/add-passmanager-module-3ae30cff52cb83f1.yaml

```diff
@@ -0,0 +1,34 @@
+---
+features:
+  - |
+    A new module :mod:`qiskit.passmanager` is added.
+    This module implements a generic pass manager and flow controllers,
+    and provides an infrastructure to manage execution of transform passes.
+    The pass manager is a baseclass and not aware of the input and output object types,
+    and you need to create a subclass of the pass manager
+    for a particular program data to optimize.
+    The :mod:`qiskit.transpiler` module is also reorganized to rebuild the existing
+    quantum circuit pass manager based off of new generic pass manager.
+    See upgrade notes for more details.
+upgrade:
+  - |
+    :class:`qiskit.transpiler.PassManager` is now a subclass of
+    :class:`qiskit.passmanager.BasePassManager`. There is no functional modification
+    due to this class hierarchy change.
+  - |
+    :mod:`qiskit.transpiler.propertyset` is moved to :mod:`qiskit.passmanager.propertyset`.
+  - |
+    Pass flow controllers :class:`~qiskit.passmanager.FlowController`,
+    :class:`~qiskit.passmanager.ConditionalController` and
+    :class:`~qiskit.passmanager.DoWhileController` are moved
+    from :mod:`qiskit.transpiler.runningpassmanager`
+    to own module :mod:`qiskit.passmanager.flow_controllers`.
```

Comment:
I'm not sure we need upgrade notes for these anymore since there is nothing users need to do now. In 0.26 when we deprecate the legacy path we can document it then.